src/h5web/visualizations/shared/AxisSystemProvider.tsx

```diff
@@ -11,24 +10,19 @@ export interface AxisInfos {
 export const AxisSystemContext = createContext<AxisInfos>({} as AxisInfos);
 
 function getAxisInfo(config: AxisConfig): AxisInfo {
-  if (isIndexAxisConfig(config)) {
-    const { indexDomain, showGrid = false } = config;
-    return {
-      isIndexAxis: true,
-      scaleFn: scaleLinear,
-      domain: indexDomain,
-      scaleType: ScaleType.Linear,
-      showGrid,
-    };
-  }
-
-  const { dataDomain, scaleType = ScaleType.Linear, showGrid = false } = config;
+  const {
+    domain,
+    scaleType = ScaleType.Linear,
+    isIndexAxis,
+    showGrid = false,
+  } = config;
+
   return {
-    isIndexAxis: false,
-    scaleFn: SCALE_FUNCTIONS[scaleType],
-    domain: dataDomain,
-    scaleType,
+    onlyIntegers: isIndexAxis,
+    scaleFn: isIndexAxis ? scaleLinear : SCALE_FUNCTIONS[scaleType],
+    scaleType: isIndexAxis ? ScaleType.Linear : scaleType,
     showGrid,
+    domain,
   };
```

Comment:
Note that `AxisInfo` and `AxisConfig` are now extremely similar: there is only `scaleFn` that can be removed as it depends on `scaleType`.

I already pointed out the fact that it was misleading to have two interfaces `Info` and `Config` so I think we can merge them in the near future.